src/log.c

```diff
@@ -525,71 +538,143 @@ void qd_log_finalize(void) {
         log_sink_free_lh(DEQ_HEAD(sink_list));
 }
 
-qd_error_t qd_log_entity(qd_entity_t *entity) {
-
+qd_error_t qd_log_entity(qd_entity_t *entity)
+{
     qd_error_clear();
 
-    //Obtain the log_source_lock global lock
-    sys_mutex_lock(log_source_lock);
-
     char* module = 0;
-    char *output = 0;
+    char *outputFile = 0;
     char *enable = 0;
+    int include_timestamp = 0;
+    int include_source = 0;
+
+    bool has_enable = false;
+    bool has_output_file = false;
+    bool has_include_timestamp = false;
+    bool has_include_source = false;
+    bool is_sink_syslog = false;
     bool trace_enabled = false;
+    bool error_in_output = false;
+    bool error_in_enable = false;
 
     do {
-
+        //
+        // A module attribute MUST be specified for a log entity.
+        // Every other attribute is optional.
+        //
         module = qd_entity_get_string(entity, "module");
+
+        //
+        // If the module is not specified, there is nothing to do, just log an
+        // error and break out of this do loop.
+        //
         QD_ERROR_BREAK();
```

Comment:
IIRC these QD_ERROR_BREAK's force a jump to the end of the while (0) loop.
At the end of the while (0) loop there is a mutex unlock call, which will be unlocking a mutex which is not locked.